presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcTypeHandle.java

```diff
@@ -47,21 +47,21 @@ public JdbcTypeHandle(
             @JsonProperty("arrayDimensions") Optional<Integer> arrayDimensions,
             @JsonProperty("caseSensitivity") Optional<CaseSensitivity> caseSensitivity)
     {
-        this(jdbcType, jdbcTypeName, columnSize, Optional.of(decimalDigits), arrayDimensions, caseSensitivity);
+        this(jdbcType, jdbcTypeName, Optional.of(columnSize), Optional.of(decimalDigits), arrayDimensions, caseSensitivity);
     }
 
     @JsonCreator
     public JdbcTypeHandle(
             @JsonProperty("jdbcType") int jdbcType,
             @JsonProperty("jdbcTypeName") Optional<String> jdbcTypeName,
-            @JsonProperty("columnSize") int columnSize,
+            @JsonProperty("columnSize") Optional<Integer> columnSize,
             @JsonProperty("decimalDigits") Optional<Integer> decimalDigits,
             @JsonProperty("arrayDimensions") Optional<Integer> arrayDimensions,
             @JsonProperty("caseSensitivity") Optional<CaseSensitivity> caseSensitivity)
     {
         this.jdbcType = jdbcType;
         this.jdbcTypeName = requireNonNull(jdbcTypeName, "jdbcTypeName is null");
-        this.columnSize = columnSize;
+        this.columnSize = requireNonNull(columnSize, "columnSize is null");
```

Comment:
consider marking this constructor as `@Deprecated`.

Reply:
This is the only non-deprecated constructor here.